File: app/src/main/java/com/example/wat2eat/models/Recipe.kt
package com.example.wat2eat.models

data class Recipe(
    val id: Int,
    val title: String,
    val image: String,
    val imageType: String,
    // TODO: ADD INGREDIENTS, DIET LABELS, HEALTH LABELS, CALORIES, ETC.
    // val readyInMinutes: Int,
    // val servings: Int,
    // val nutrition: Nutrition
)

data class Nutrition(
    val nutrients: List<Nutrient>
)

data class Nutrient(
    val name: String,
    val amount: Double,
    val unit: String,
    val percentOfDailyNeeds: Double
)

File: backend/src/db/recipe.ts
import mongoose from 'mongoose';

export const RecipeSchema = new mongoose.Schema({
	id: { type: String },
	title: { type: String },
	image: { type: String },
	servings: { type: Number },
	readyInMinutes: { type: Number },
	extendedIngredients: { type: Array },
	analyzedInstructions: { type: Array },
});

export const Recipe = mongoose.model('Recipe', RecipeSchema);
